pkg/parser/parser.go

```diff
@@ -342,13 +344,26 @@ func getServiceEndpoints(log logrus.FieldLogger, s store.Storer, svc corev1.Serv
 	return targetsForEndpoints(endpoints)
 }
 
+// getIngressClassParamsOrDefault returns the parameters for the current ingress class.
+// If the cluster operators have specified a set of parameters explicitly, it returns those.
+// Otherwise, it returns a default set of parameters.
+func getIngressClassParamsOrDefault(s store.Storer) *configurationv1alpha1.IngressClassParamsSpec {
```

Comment:
I'd like this function to make a clear distinction between the "not found" case and the "runtime error" case.

I suggest the following return values:
```go
params, nil // in the "found" case
nil, nil // in the "not found" case
nil, err // in case of the "runtime error" case
```